rocAL/rocAL/source/readers/image/external_source_reader.cpp

```diff
@@ -0,0 +1,264 @@
+    // _curr_file_idx++;
+        _curr_file_idx = (_curr_file_idx + 1) % _batch_count;
+}
+
+size_t ExternalSourceReader::open()
+{
+    if (_mode == FileMode::FILENAME) {
+        std::string next_file_name;
+        bool ret = pop_file_name(next_file_name);   // Get next file name: blocking call, will wait till next file is received from external source
+        if (_end_of_sequence && !ret) // shobi check this
+          return 0;
+        // prefix with folder_path of exists
+        if (!_folder_path.empty())
+        //   next_file_name = _folder_path + "/" + next_file_name;
+            next_file_name = next_file_name;
+        _last_id= next_file_name;
+        filesys::path pathObj(next_file_name);
+        if(filesys::exists(pathObj) && filesys::is_regular_file(pathObj))
+        {
+          _current_fPtr = fopen(next_file_name.c_str(), "rb");// Open the file,
+          if(!_current_fPtr) // Check if it is ready for reading
+              return 0;
+
+          fseek(_current_fPtr, 0 , SEEK_END);// Take the file read pointer to the end
+
+          _current_file_size = ftell(_current_fPtr);// Check how many bytes are there between and the current read pointer position (end of the file)
+
+          if(_current_file_size == 0)
+          { // If file is empty continue
+              fclose(_current_fPtr);
+              _current_fPtr = nullptr;
+              return 0;
+          }
+          fseek(_current_fPtr, 0 , SEEK_SET);// Take the file pointer back to the start
+          _file_data[_curr_file_idx] = std::make_tuple((unsigned char *)next_file_name.data(), (size_t)_current_file_size, 0, 0, 0); // shobi check typecasting
+          incremenet_read_ptr();
+        }
+    } else {
+        std::tuple<unsigned char*, size_t, int, int, int> image;
+        bool ret = pop_file_data(image);
+        if (_end_of_sequence && !ret) // shobi check this
+        {
+            std::cerr<<"\n !!!!!!!!!!!!!!!!!! EOS || POP FAILED !!!!!!!!!!!!!!!!!!!!";
+            return 0;
+        }
+        _file_data[_curr_file_idx] = image;
+        _current_file_size = std::get<1>(image);
+    }
+
+    return _current_file_size;
+}
+
+size_t ExternalSourceReader::read_data(unsigned char* buf, size_t read_size)
+{
+    if (_mode == FileMode::FILENAME) {
+        if(!_current_fPtr)
+            return 0;
+
+        // Requested read size bigger than the file size? just read as many bytes as the file size
+        read_size = (read_size > _current_file_size) ? _current_file_size : read_size;
+
+        size_t actual_read_size = fread(buf, sizeof(unsigned char), read_size, _current_fPtr);
+        return actual_read_size;
+    } else {
+        unsigned char *file_data_ptr = std::get<0>(_file_data[_curr_file_idx]);
+        size_t size = _current_file_size;
+        incremenet_read_ptr();
+        if (size > read_size)
+          THROW("Requested size doesn't match the actual size for file read")
+        memcpy(buf, (void *)file_data_ptr, size);
```

Comment:
Please change c-style casting to static_cast, applicable to all such instances in new files added